apps/snapodds/src/app/odds/odds.component.ts

```diff
@@ -37,23 +36,22 @@ export class OddsComponent implements OnChanges {
     this.loading = true;
     this.noResults = false;
 
-    this.snapOddsFacade
-      .getLineOdds(sportEventId)
-      .pipe(
-        tap((lineOdds) => {
-          if (lineOdds.sportsBooks?.length === 0) throw new Error('No SportsBooks found');
-        })
-      )
-      .subscribe({
-        next: (lineOdds) => {
-          this.lineOdds = lineOdds;
-          this.loading = false;
-        },
-        error: () => {
+    this.snapOddsFacade.getLineOdds(sportEventId).subscribe({
+      next: (lineOdds) => {
+        if (lineOdds.sportsBooks == undefined || lineOdds.sportsBooks.length === 0) {
```

Comment:
I don't understand the benefits of this change. 
- Please use optional changing as it was before (you can read upon it here https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html). 
- By having this in the "success" event you've duplicated the lines 51:53 from the error block. If you don't like to throw an error at this point, then please move these 3 lines into a function and call it at #42 and #51.

Reply:
Optional chaining is not correct in this case, because actually from API we receive a response without sportsBooks collection (it is not an empty array, it is undefined) and in this case component does not show an empty message. I have added corresponding test case and introduced a function as you suggested.